torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -731,6 +745,56 @@ TensorView* clamp(TensorView* in, Val* min_val, Val* max_val) {
   return clamp(in->as<Val>(), min_val, max_val)->as<TensorView>();
 }
 
+// sum_to operator
+
+// Name of sum_to is different from NV fuser naming,
+// this is to align with the operator name
+TensorView* sum_to(TensorView* v1, const std::vector<Int*>& shape) {
+  const auto& v1_root = TensorDomain::noReductions(v1->getRootDomain());
+
+  TORCH_CHECK(
+      v1_root.size() >= shape.size(),
+      "sum_to: Error trying to reduce",
+      v1,
+      "into a shape of size",
+      shape.size());
+
+  // If no reduction is needed sum_to returns the input tv
+  TensorView* v2 = v1;
+
+  const int64_t leading_dims = v1_root.size() - shape.size();
+
+  // Generate reduction axes for leading dims
+  std::vector<int> reduce_dims(leading_dims);
+  std::iota(reduce_dims.begin(), reduce_dims.end(), 0);
+
+  // Generate reduction axes for dims within shape
+  std::vector<bool> inner_red_dims(shape.size(), false);
+  bool reduction_within_shape = false;
+
+  // Reduce rest of the dims with keep_dim
+  for (int i = leading_dims; i < v1_root.size(); i++) {
+    if (shape[i - leading_dims]->isOneInt() &&
+        !v1_root[i]->rawExtent()->isOneInt()) {
+      inner_red_dims[i - leading_dims] = true;
+      reduce_dims.push_back(i);
+      reduction_within_shape = true;
+    }
+  }
+
```

Comment:
assert here that reduce_dims.size() == v1_root.size() ?

Reply:
I believe this is not necessarily always true. reduce_dims contains all the dims that need to be reduced by sum_to, while v1_root has all the dim's of input tensor. Eg:
```cpp
  tv1 = T1[i0(3),i1(4),i2(5),i3(6)];
  tv2 = sum_to(tv1,{5,1});
```
leads to 
```cpp
reduce_dim = {0,1,3}
v1_root  = {i0(3),i1(4),i2(5),i3(6)} (actually a vector of iterdomains but showing the sizes only)
```

Reply:
Makes sense, thanks